lnrpc/routerrpc/router.proto

```diff
@@ -502,6 +502,36 @@ message MissionControlConfig {
     before we raise the failure amount.
     */
     uint64 minimum_failure_relax_interval = 5;
+
+    enum ProbabilityModel {
+        UNKNOWN = 0;
+        APRIORI = 1;
+        BIMODAL = 2;
```

Comment:
How would things look if instead of adding all estimators into the core lnd code base, they could live in an external process that interfaces with a lean lnd instance?

Reply:
> How would things look if instead of adding all estimators into the core lnd code base, they could live in an external process that interfaces with a lean lnd instance?

Good q, I think we're pretty much on this path now. If we take this current `Estimator` interface, then add something like a `EstimatorMemory` interface (abstraction over mission control), we can expose something like:
```
type BayesianEstimator interface {
    Estimator 
    MissionControl
}
```
 
or w/e 

This can then be exposed in the top level lnd config like some of the persistence stuff already is. If set, then lnd would just use that vs making an estimator based on the CLI args. 

This mission control cfg RPC is pretty low level, and should be considered experimental and more of an exception to the path painted above. This would just be for estimators that lnd knows of first hand. Ofc, it can also be extended to pass in estimator configs it doesn't know about directly to the estimator to hot swap.

Reply:
Will try to implement something along those lines, extracting that functionality sounds like a good approach (mission control and the probability estimator can be viewed as a logical unit). I'd only see issues in making probability queries efficient, but don't have any experience in what cross-process communication would add there.

Reply:
> Will try to implement something along those lines, extracting that functionality sounds like a good approach

I think for the scope of this PR, it's enough just to expose the _existing_ interface from the top-level config. Then we can spin off a new issue to brainstorm what a functionally complete interface would look like.